x/campaign/keeper/grpc_auctions_by_campaign.go

```diff
@@ -0,0 +1,38 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	fundraisingtypes "github.com/tendermint/fundraising/x/fundraising/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/tendermint/spn/x/campaign/types"
+)
+
+func (k Keeper) AuctionsOfCampaign(goCtx context.Context, req *types.QueryAuctionsOfCampaignRequest) (*types.QueryAuctionsOfCampaignResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	// not performing any existence check on the campaign as it's not needed
+	reqCampaignID := req.CampaignID
+	resAuctionIDs := make([]uint64, 0)
+
+	k.fundraisingKeeper.IterateAuctions(ctx, func(auction fundraisingtypes.AuctionI) (stop bool) {
```

Comment:
We should also add that the auctioneer is the coordinator of the campaign asa condition

Reply:
It cannot work. If the auction has been created and the coordinator is updated, it would break. I thought about it and I think it could break very easily as a check

Reply:
Would be enforceable as a check only if we had the coordinator ID (not the coordinator address) stored in the auction data

Reply:
True

Maybe it's fine to not have this check if we can remove this query in the near future depending on advancement on the backend service

Reply:
We might also use the `created` [hooks](https://github.com/tendermint/fundraising/blob/main/x/fundraising/keeper/auction.go#L68) to:
1. get the campaign from the selling denom
2. check that at creation the coordinator of the campaign is the auction creator (so we can even prevent auctions creation if the coordinator is not the auctioneer)
3. store a direct mapping of `auctionIDs` to `campaignID` so fetching this info might be extremely faster

Reply:
but this would widen the scope of this PR, and change the logic a bit. So LMK your thoughts

Reply:
As discussed in the call. Let's omit coordinator verification for now